tgui/packages/tgui/interfaces/PersonalCrafting.tsx

```diff
@@ -230,6 +233,11 @@ export const PersonalCrafting = (props, context) => {
   const content = document.getElementById('content');
   const CATEGORY_ICONS =
     mode === MODE.cooking ? CATEGORY_ICONS_COOKING : CATEGORY_ICONS_CRAFTING;
+  const searchRecipes = (value) => {
+    setPages(1);
+    setSearchText(value);
+  };
+  const debouncedSearch = debounce(searchRecipes, 500);
```

Comment:
This won't work as expected, because UI function is re-run many times when re-rendering, every time it will be a new debouncedSearch function and onInput instead of calling just one function many times will call many functions once, making debounce pointless.

You should rewrite it in such a way so that it is declared outside of the UI, then it will stay the same across re-renders and debounce will work. It won't have access to `setPages` and `setSearchText` though, but you can pass them as arguments.